Add BmiClientApptainer

sverhoeven opened the pull request:
Adds `grpc4bmi.bmi_client_apptainer.BmiClientApptainer` class. 

Fixes #116

Requires a https://apptainer.org installation to run tests.

After this PR is merged a new version on the 0.2.x branch should be published to GitHub/PyPi/Zenodo.

Due to Singularity and Apptainer being so similar, the code duplication has increased, this will disappear once we deprecate singularity.

BSchilperoort commented on grpc4bmi/bmi_client_apptainer.py:
```diff
@@ -0,0 +1,273 @@
+    .. code-block:: python
+
+        from grpc4bmi.bmi_client_apptainer import BmiClientApptainer
+        client = BmiClientApptainer(image='docker://ewatercycle/marrmot-grpc4bmi:latest')
```
This line won't run. work_dir is a required argument;
`TypeError: __init__() missing 1 required positional argument: 'work_dir'`

BSchilperoort replied:
The next 3 lines run fine if a work_dir is given.

BSchilperoort commented on grpc4bmi/bmi_client_docker.py:
```diff
@@ -23,7 +23,7 @@ class BmiClientDocker(BmiClient):
         input_dirs (Iterable[str]): Directories for input files of model.
 
-            All of directories will be mounted read-only inside Singularity container on same path as outside container.
+            All of directories will be mounted read-only inside Docker container on same path as outside container.
```
Well spotted! :)

BSchilperoort reviewed (approved):
Hi Stefan, very nice work. I was able to run the (`test_apptainer`) tests locally (on WSL2). The design overview graph looks a lot better 👍 

I did notice a small mistake in a docstring example though (see the specific comment), but I'll approve the PR already.

Additionally I wondered about the docs generation, it seems that the readthedocs CI is not linked to this repository? It makes going through the API docstring a bit harder, and perhaps some of the docs changes break rtd.

sverhoeven commented:
Thanks for reviewing. 

I fixed the docstring example, it was also present in the BmiClientSingularity docstring.

The https://grpc4bmi.readthedocs.io/en/0.2.x/ gets updated each 0.2.x merge. I updated the docs badge to point to that.

sverhoeven commented:
v0.2.16 has been released.
